samples/drivers/gecko_adc/src/battery_ctrl.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2022 T-Mobile USA, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <zephyr/devicetree.h>
+
+#if DT_NODE_EXISTS(DT_NODELABEL(bq24250))
+#include "bq24250.h"
+#endif
+
+#include "battery_ctrl.h"
+
+#define TABLE_LEN 11
+#define TABLE_VOLTAGE 0
+#define TABLE_PERCENT 1
+
+/* 100% value based on actual new battery measurement */
+static float battery_discharging_tbl[TABLE_LEN][2] = {
+	{4.145, 100.0 },
+	{4.072, 90.0  },
+	{3.992, 80.0  },
+	{3.923, 70.0  },
+	{3.858, 60.0  },
+	{3.806, 50.0  },
+	{3.775, 40.0  },
+	{3.752, 30.0  },
+	{3.729, 20.0  },
+	{3.686, 10.0  },
+	{3.632,  0.0  }};
+
+/* 100% value based on actual new battery measurement */
+static float battery_charging_tbl[TABLE_LEN][2] = {
+	{4.175, 100.0 },
+	{4.130, 90.0  },
+	{4.071, 80.0  },
+	{4.005, 70.0  },
+	{3.934, 60.0  },
+	{3.886, 50.0  },
+	{3.853, 40.0  },
+	{3.833, 30.0  },
+	{3.800, 20.0  },
+	{3.741, 10.0  },
+	{3.597,  0.0  }};
```

Comment:
The preceding two-dimensional arrays would be better implemented as single-dimensional arrays of struct. This would eliminate the need for the TABLE_LEN, TABLE_VOLTAGE, and TABLE_PERCENT macros and provide additional flexibility in the size of the tables. For example:
```
struct  {
float voltage, percent;
} battery_discharging_tbl[] = {
	{4.145, 100.0 },
 	{4.072, 90.0  },
 	{3.992, 80.0  },
 	{3.923, 70.0  },
 	{3.858, 60.0  },
 	{3.806, 50.0  },
 	{3.775, 40.0  },
 	{3.752, 30.0  },
 	{3.729, 20.0  },
 	{3.686, 10.0  },
 	{3.632,  0.0  }
}, battery_charging_tbl[] = {
{
	{4.175, 100.0 },
 	{4.130, 90.0  },
 	{4.071, 80.0  },
 	{4.005, 70.0  },
 	{3.934, 60.0  },
 	{3.886, 50.0  },
 	{3.853, 40.0  },
 	{3.833, 30.0  },
 	{3.800, 20.0  },
 	{3.741, 10.0  },
 	{3.597,  0.0  }
 	};
```

Reply:
This is a rather subtantial change.  Maybe it should be brought up with the author.  I believe John wrote this.

Reply:
I agree.